### PR TITLE
Exam registration process

### LINKED ISSUE #1541: Exam registration process
Currently, the platform does not support any primitive for registering for an exam. 

This is cumbersome, since it means that registration process - the act of declaring interest for taking an exam at a given date - must be carried outside the platform, and results must then be manually evaluated. We should instead allow registration to take place in the platform.

The flow should be like the following: 

1. An organization janitor must create an `ExamRegistration`, which is tied to an organization - or event a course eventually - and one or more future  exams - which should have been created previously. The exam registration should have also a `description` that will be displayed to students, and a `start` and `end` period - exam registration will not popup before the start datetime and will be unusable after the `end` datetime. It also will have an `authorization_criteria` (or condition). Read along for more details about this. 
2. When the exam registration starts, users in the organization must be notified with an in-app message. It should be displayed in the profile's bell. 
3. When users enters the notification, it should redirect - or contains a text with a link - to a new page in `laboratory` that allows they to register for the exam, by choosing an `exam_appointment` from the available exams at the `ExamRegistration`. Although each appointment is actually an `Exam` object, it should be represented as just a time slot. Since there is currently a direct mapping from `Exam`s to `ExamAppointment`s, I think there is no need here to introduce a new `ExamAppointment` class.   
4. When the ExamRegistration ends, we must run a batch process that automatically creates permissions - in the same manner classroom-api allows to register students for an exam - according to the exam appointments. @julian-berbel has already done things like those. 
5. Not all registered students will be actually authorized to the real exam object. This aspect will be controlled by the `authorization_criteria`. We expect by now to support the following restrictions: 
    * every registered student is authorized when appropriate
    * only registered students that have a given amount of passed exercises at the end time will be authorized. We should use the `top_status` of the assignment **in the organization**,  not the current status, in order students to allow safe edition of exercises after they have solved.  In order to be consistent with classroom reports, we should at the same time fix this issue: https://github.com/mumuki/mumuki-classroom-api/issues/236
    * a particular content has been solved - chapter, lesson or book.
6. After job is run, student must receive a confirmation of authorization, or a deny message, in the same way as item 2. 

Item 1 should ideally have a UI and/or a classroom API, but for now having a model support for is just fine. However, the remaining phases must be fully implemented. 

Item 2 should also actually fire a notification by email. I think that may be part of a separate issue, that consist of enable all kind of notifications to produce mails if activated at organization and/or user level.

### PR DESCRIPTION
**Pull request by faloi**
Closes #1541 

### Notifications panel

From top to bottom: 
* new message in discussion, 
* new direct message, 
* exam registration start.

image

From top to bottom: 
* exam authorization request updated (approval / rejection not shown on purpose), 
* new direct message.

image

### Exam registration

image

### Exam authorization request

image
image

### CODE REVIEW

File: lib/mumuki/laboratory/locales/en.yml [hunk 1]
```diff
@@ -11,88 +13,112 @@ en:
...
+  discussion_new_message: New message on %{title}
   discussion_updated: Discussion updated
   discussions: Discussions
   dont_leave_us: Don't leave us! Learning is fun. You just have to keep at it.
   download: Download your solution
   edit: Edit
   edit_profile: Edit profile
-  editor_placeholder: "write your solution here..."
+  editor_placeholder: write your solution here...
+  eeee: s
```
Comment (felipecalvo) [inline lib/mumuki/laboratory/locales/en.yml:88]: We might not be using this one anytime soon :stuck_out_tongue:

Comment (felipecalvo) [inline lib/mumuki/laboratory/locales/en.yml:88]: We might not be using this one anytime soon :stuck_out_tongue:

Comment (faloi) [reply lib/mumuki/laboratory/locales/en.yml:88]: Seems that you don't understand the "eeee" language... 🤣

File: app/controllers/exam_authorization_requests_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  def update
+    ExamAuthorizationRequest.update params[:id], authorization_request_params
+    send_updated_notice!
+    redirect_to root_path
+  end
+
+  private
+
+  def send_updated_notice!
```
Comment (luchotc) [inline app/controllers/exam_authorization_requests_controller.rb:22]: Just nitpicking here, but this method name is a bit misleading. We could just have to two different notices, one for request created and another for updated and avoid this method altogether

File: add_translation.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env ruby
```
Comment (luchotc) [inline add_translation.rb:1]: Let's keep this one out, I'll add it to another repository which is more suitable for this type of scripts

**Review by luchotc — APPROVED** [review]
It looks great :heart: 
I made some truly minor comments and we are all set.

Let's please add some feature tests for this one though!

File: add_translation.rb (hunk 3 above)
Comment (faloi) [reply add_translation.rb:1]: Fine, I'll rebase and delete the commit that adds it. I've sent you the code via WhatsApp. 🙈

**Pull request merged**